java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/utils/DateTimeUtils.java

```diff
@@ -36,21 +38,17 @@ private DateTimeUtils() {
   }
 
   /**
-   * Subtracts given Calendar's TimeZone offset from epoch milliseconds.
+   * Apply calendar timezone to epoch milliseconds.
    */
   public static long applyCalendarOffset(long milliseconds, Calendar calendar) {
```

Comment:
Hmm, looking at how it's used, now I'm thinking both implementations are wrong. For instance, it's used in DateVectorAccessor and is applied to the value from the vector and the user-supplied calendar. This function appears to _assume the given milliseconds value is a naive timestamp in the given (or default) timezone_, and converts it to a UTC timestamp. **This is backwards.** The value from the Arrow vector is ALREADY a UTC timestamp!

That said, this function does seem to do the right thing for _other_ places where it is used. So I think we need to name and properly document it, and remove it from places where it should NOT be applied.